articles/advanced-r-concepts/index.md

```diff
@@ -0,0 +1,333 @@
+5        e        E        5       10    FALSE
+```
+
+R provides several functions to understand and analyze dataframes. Analysis of dataframes is necessary to perform pre-processing on the dataset. There are primarily two types of analysis: Statistical and Categorical.
+
+The statistical analysis deals with all the continuous values (numbers) in the dataset, whereas categorical analysis deals with categories.
+
+Examples for the two are as follows:
+1. Statistical values: Prices of houses, Oxygen levels in the blood, etc.
+2. Categorical values: Names of places or objects, status(example: Sell, Rent, Buy)
+
+R provides a few functions to help analyze dataframes:
+1. head(): Prints out the first six observations of the dataframe
+2. str(): Prints the structure of the dataframe. The structure includes information about the dataframe and the data-types present in each vector. Observe that the dataframe auto-assigns the data type to the vectors.
+3. tail(): Prints out the last six observations of the dataframe.
+
+The outputs for the functions `head` and `tail` are the same in this case, for lack of elements.
+
+```txt
+    vector_1 vector_2 vector_3 vector_4 vector_5
+1        a        A        1        2     TRUE
+2        b        B        2        4     TRUE
+3        c        C        3        6     TRUE
+4        d        D        4        8    FALSE
+5        e        E        5       10    FALSE
+```
+
+These functions make sense when the dataframe size is enormous. The output for the `str` function is given below:
+
+```r
+data.frame':    5 obs. of  5 variables:
+$ vector_1: Factor w/ 5 levels "a","b","c","d",..: 1 2 3 4 5
+$ vector_2: Factor w/ 5 levels "A","B","C","D",..: 1 2 3 4 5
+$ vector_3: num  1 2 3 4 5
+$ vector_4: num  2 4 6 8 10
+$ vector_5: logi  TRUE TRUE TRUE FALSE FALSE
+```
+Observe that `vector_1` is assigned to a factor data-type. To brush up on factor data-type, revisit the previous article.
+
+Dataframes support indexing. The intuition behind indexing is similar to that of the matrices. Let us consider a few examples:
+
+1. Select the third column
+2. Select the first three rows
+3. Select the first three rows from the fifth column
+
+The subsequent outputs is given below:
+
+```r
+# Consider the same data_frame defined above: new_data_frame
+new_data_frame[,3]
+new_data_frame[1:3,]
+new_data_frame[1:3,5]
+```
+
+The above operations output:
+
+```txt
+    1 2 3 4 5
+
+vector_1 vector_2 vector_3 vector_4 vector_5
+1        a        A        1        2     TRUE
+2        b        B        2        4     TRUE
+3        c        C        3        6     TRUE
+
+    TRUE TRUE TRUE
+```
+
+The final function under dataframes is `subset.` Subset function is similar to Python's filter function. We can specify conditions using subset and therefore get a slice of the dataframe satisfying the condition. **Let the condition be all those rows where `vector_5` is `not equal` to `FALSE.`**
```

Comment:
This is misleading since it gives the impression that this is the last available function for data frames, when there are many more. You can say "A final function to consider is 'subset'

Link to Python's doc for filter.

Example: Let the condition be... on a separate line